### PR TITLE
HPCC-31459 Generate GitHub Action chain diagram from action (.yml) files

### PR DESCRIPTION
**Pull request by Charan-Sharan**
HPCC-31459 Generate GitHub Action chain diagram from action (.yml) files
This Action workflow generates a markdown file that contains useful graphs that give a brief overview of all the GitHub Actions Workflows.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by AttilaVamos — CHANGES_REQUESTED** [review]
Can you add:
- condition for running only if action file changed
- execute as a workflow item

File: .github/workflows/Actions-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+          yaml_files=glob.glob(workflowsDir+'*.y*ml')
+          workflowDetails={}  #This is the main dictionary where all the workflow details are stored
+          
+          for file in sorted(yaml_files):
+            file=file.split('/')[-1]
+            getWorkflowName(workflowsDir,file)
+          
+          markdown=workflowsDir+'ActionFlow.md'
+
+          with open( markdown ,'a') as markdownFile:
```
Comment (AttilaVamos) [inline .github/workflows/Actions-workflow.yml:181]: Why do you open the markdown file for append? With that mode every time the markdown file might grow and hold old diagrams as well.

File: .github/workflows/Actions-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                  inputs=inputs.replace('\"','')    #remove " (double quotes)
+                  inputs=inputs.replace("\'",'')    #remove ' (single quotes)
+                  if len(inputs)!=0:
+                    markdownFile.write(f"\n  {job}(\"{job}\ninputs:[{inputs}]\") ---> {uses}(\"{workflowDetails[uses]['name']}[{uses}]\")")
+                  else:
+                    markdownFile.write(f"\n  {job}({job}) ---> {uses}")
+              
+              markdownFile.write('\n\n```\n\n')
+        
+          with open( markdown ,'a') as markdownFile:
```
Comment (AttilaVamos) [inline .github/workflows/Actions-workflow.yml:219]: Why are you reopen the markdown file? It is already opened in line 181.

File: .github/workflows/Actions-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+                    
+            for trigger in sorted(triggersList):
+              markdownFile.write('\n```mermaid\n')
+              markdownFile.write('\nflowchart TB\n')
+              markdownFile.write(f'\n  subgraph {trigger.upper()}')
+              for file in sortedDictionary:
+                if trigger in workflowDetails[file]['trigger']:
+                  markdownFile.write(f'\n    {trigger} ---> {file}[\"{workflowDetails[file]["name"]}[{file}]\"]')
+              markdownFile.write('\n  end\n')
+              markdownFile.write('\n```\n')
```
Comment (AttilaVamos) [inline .github/workflows/Actions-workflow.yml:234]: The close is missing for the markdown file. It is a good practice if you open a file then close when not used anymore.

**Review by AttilaVamos — CHANGES_REQUESTED** [review]
It is almost there, only a couple of questions, requests.

File: .github/workflows/Actions-workflow.yml (hunk 1 above)
Comment (Charan-Sharan) [reply .github/workflows/Actions-workflow.yml:181]: On line 30, I will clear the file. So, all the previous graphs will be erased.

Comment (AttilaVamos) [reply .github/workflows/Actions-workflow.yml:181]: Yes, you are right.

**Review by AttilaVamos — APPROVED** [review]
Seems good.

File: .github/workflows/Actions-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        shell: python
```
Comment (Michael-Gardner) [inline .github/workflows/Actions-workflow.yml:223]: Just for cleanliness, can we move this above the run: without breaking anything?  Other than that, everything looks great.

**Review by Michael-Gardner — APPROVED** [review]
Approving

File: .github/workflows/Actions-workflow.yml (hunk 4 above)
Comment (Charan-Sharan) [reply .github/workflows/Actions-workflow.yml]: Yes, it is possible to move it above run:

**Comment by Charan-Sharan** [comment]
@ghalliday  could you please merge this PR?

**Pull request merged**